### DISCUSSION TITLE
Questions about the outreachy contributions

### DISCUSSION THREAD

**Discussion by mina-atef-00** [LesFruitsDefendus/saskatoon-ng#130] [Outreachy]
Hello my name is Mina, an outreachy applicant with experience in python and django, and interested in and would be happy to be contributing to the project.

I have some questions could you please answer them:

- What is the preferred way to communicate with the mentor?
- What are the contribution guidelines, naming convention, etc...
- What versions of python, django recommended? Any other libraries, programs, containers I will need to install other than the ones provide in the INSTALL.md?
- What issues should I start with first from the ones stated on outreachy? How will the issues assignments go? Do I say which issue I will work on or will you assign me issues?
- Should I use mysql or sqlite3 for development? Which is recommended?

**Comment by tristanlatr** [comment]
Hello @mina-andajos,

Glad to see that your are interested into Saskatoon, I'll try to answer all of your questions. 

> What is the preferred way to communicate with the mentor?

  Github is good, but I'm pretty flexible.

> What are the contribution guidelines, naming convention?

  Currently we don't have any of these clearly defined, though following PEP8 is recommended everywhere. Please note that util PR #114 is merged, all new pull requests should target the ``develop`` branch.  

> What versions of python, django recommended? 

Python3.8 and Django `Django>=3.0.14,<4.0.0` (as you figured out). Every other thing should be covered in the INSTALL.md

> What issues should I start with first from the ones stated on outreachy?

The issues stated in the outreachy project description are for the whole internship, so you shoun't start with those issues for the contribution period.

> How will the issues assignments go? 

We'll talk about what are the priorities and what the intern feels like capable of doing. Set some objectives for the internship. The intern will meet every week with me and other people from the collective to follow-up on objectives advancement. We will not micro-manage you ;-)

> Do I say which issue I will work on or will you assign me issues?

As described above, it's mix of both. But for sure I won't "assign" intern issue but rather propose to work on this or that issue, and the intern will have a word to say.

> Should I use mysql or sqlite3 for development? Which is recommended?

We've been using sqlite3 for development.

I hope I've answered your questions :)

Talk to you later,

Tristan

**Reply by paulinebanye** [reply]
Hi @tristanlatr this was actually quite detailed and it answered some questions I had as well. Thank you

**Reply by tristanlatr** [reply]
Glad it helped!

Talk to you later,

Tristan

**Reply by ojaragodwin** [reply]
Hi i had similar questions but i am glad they were all answered clearly. 
Thank you @mina-andajos
